packages/views.vite-plugin/README.md

```diff
@@ -0,0 +1,100 @@
+# @osdk/views-manifest-vite-plugin
+
+This is a vite plugin that will automatically discover `*.config.(j|t)s` files that are imported into entrypoint files and generates a `.palantir/view-config.json` file for Foundry to read from. This allows developers to write out their configuration for custom views in Foundry in a single, type-safe file and not worry about generating a manifest file needed for publishing new versions of the view within Foundry.
+
+## Usage
+
+To use the plugin, add it to the `plugins` list of your [vite configuration](https://vitejs.dev/config/):
+
+```js
+import viewManifestPlugin from "@osdk/views-manifest-vite-plugin";
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  plugins: [viewManifestPlugin()],
+  // Rest of configuration
+});
+```
+
+## Defining configuration
+
+The `@osdk/views-client` package exports a `ViewConfig` type so that you can define a configuration object in the shape that this plugin requires. You can define your config like so:
+
+```js
+// main.config.ts
+export default {
+  rid: "<FILL IN RID>", // This is the RID of the Foundry view this code will publish to
+  parameters: {
+    headerText: {
+      displayName: "Widget title",
+      type: "string",
+    },
+    showWarning: {
+      displayName: "Show warning callout",
+      type: "boolean",
+    },
+    todoItems: {
+      displayName: "Todo items",
+      type: "array",
+      subType: "string",
+    },
+  },
+  events: {
+    updateHeader: {
+      displayName: "Update header",
+      parameterIds: ["headerText"],
+    },
+    updateTodoItems: {
+      displayName: "Update todo items",
+      parameterIds: ["todoItems"],
+    },
+  },
+} as const satisfies ViewConfig;
+```
+
+## Importing the configuration
+
+Import your `*.config.ts` file in any entrypoint JavaScript/TypeScript file so the plugin can pick it up. Entrypoint files are any that are imported from your HTML file. For example, if you have the following setup:
+
+```
+src/
+  main.tsx
+  main.config.ts
+index.html
```

Comment:
Can you give an example file structure of what you think a multiple view publishing project would look like to make sure I understand? For example, do they each have an index.html (is this only used for discovering entrypoints?) or are they all just script tags in a single dummy index.html

Reply:
yup can do! for posterity they would each have their own index.html (since that's how vite works)

Reply:
hm... im viewing this vite plugin as an atomic unit that can be composed no matter what the rest of your vite config may look like - i think if we wanted to wrap it in an opinionated config i would do that elsewhere.